client/main-layout.tsx

```diff
@@ -115,315 +118,284 @@ const FadedSettingsIcon = styled(SettingsIcon)`
-    } else if (!isStarcraftHealthy(this.props)) {
-      this.props.dispatch(openDialog(DialogType.StarcraftHealth))
+  const onCreateLobbyClick = () => {
+    if (!isShieldBatteryHealthy({ starcraft })) {
+      dispatch(openDialog(DialogType.ShieldBatteryHealth))
+    } else if (!isStarcraftHealthy({ starcraft })) {
+      dispatch(openDialog(DialogType.StarcraftHealth))
     } else {
-      this.props.dispatch(openOverlay('createLobby'))
+      // TODO(2Pac): Remove `any` once the `openOverlay` is TS-ified
+      dispatch(openOverlay('createLobby') as any)
     }
   }
 
-  onJoinLobbyClick = () => {
-    if (!isShieldBatteryHealthy(this.props)) {
-      this.props.dispatch(openDialog(DialogType.ShieldBatteryHealth))
-    } else if (!isStarcraftHealthy(this.props)) {
-      this.props.dispatch(openDialog(DialogType.StarcraftHealth))
+  const onJoinLobbyClick = () => {
+    if (!isShieldBatteryHealthy({ starcraft })) {
+      dispatch(openDialog(DialogType.ShieldBatteryHealth))
+    } else if (!isStarcraftHealthy({ starcraft })) {
+      dispatch(openDialog(DialogType.StarcraftHealth))
     } else {
-      this.props.dispatch(openOverlay('joinLobby'))
+      // TODO(2Pac): Remove `any` once the `openOverlay` is TS-ified
+      dispatch(openOverlay('joinLobby') as any)
     }
   }
 
-  onMapUpload = map => {
-    this.props.dispatch(
-      openOverlay('browseServerMaps', { ...this.serverMapsProps, uploadedMap: map }),
-    )
+  const onMapUpload = (map: MapInfoJson) => {
+    // TODO(2Pac): Remove `any` once the `openOverlay` is TS-ified
+    dispatch(openOverlay('browseServerMaps', { ...serverMapsProps, uploadedMap: map }) as any)
   }
 
-  onMapDetails = map => {
-    this.props.dispatch(openDialog(DialogType.MapDetails, { mapId: map.id }))
+  const onMapDetails = (map: MapInfoJson) => {
+    dispatch(openDialog(DialogType.MapDetails, { mapId: map.id }))
   }
 
-  onRemoveMap = map => {
-    this.props.dispatch(removeMap(map))
+  const onRemoveMap = (map: MapInfoJson) => {
+    dispatch(removeMap(map))
   }
 
-  onRegenMapImage = map => {
-    this.props.dispatch(regenMapImage(map))
+  const onRegenMapImage = (map: MapInfoJson) => {
+    dispatch(regenMapImage(map))
   }
 
-  serverMapsProps = {
+  const serverMapsProps = {
     title: 'Maps',
-    onMapUpload: this.onMapUpload,
-    onMapSelect: this.onMapDetails,
-    onMapDetails: this.onMapDetails,
-    onRemoveMap: this.onRemoveMap,
-    onRegenMapImage: this.onRegenMapImage,
+    onMapUpload,
+    onMapSelect: onMapDetails,
+    onMapDetails,
+    onRemoveMap,
+    onRegenMapImage,
   }
 
-  onMapsClick = () => {
-    if (!isShieldBatteryHealthy(this.props)) {
-      this.props.dispatch(openDialog(DialogType.ShieldBatteryHealth))
-    } else if (!isStarcraftHealthy(this.props)) {
-      this.props.dispatch(openDialog(DialogType.StarcraftHealth))
+  const onMapsClick = () => {
+    if (!isShieldBatteryHealthy({ starcraft })) {
```

Comment:
it is incredibly weird to me that we require a valid installation to browse server maps so... yeah. I'm gonna leave it as is for now because that's how it was but I mean, whatever reason we have for doing this here, we should fix that